src/expr.rs

```diff
@@ -1210,6 +1211,28 @@ fn rewrite_string_lit(context: &RewriteContext<'_>, span: Span, shape: Shape) ->
     )
 }
 
+fn rewrite_int_lit(context: &RewriteContext<'_>, lit: &ast::Lit) -> Option<String> {
+    let span = lit.span;
+    let symbol = lit.token.symbol.as_str();
+
+    if symbol.starts_with("0x") {
+        let hex_lit = match context.config.hex_literal_case() {
+            HexLiteralCase::Ignore => None,
+            HexLiteralCase::ToUpper => Some(symbol[2..].to_ascii_uppercase()),
+            HexLiteralCase::ToLower => Some(symbol[2..].to_ascii_lowercase()),
+        };
+        if let Some(hex_lit) = hex_lit {
+            return Some(format!(
+                "0x{}{}",
+                hex_lit,
+                lit.token.suffix.map_or(String::new(), |s| s.to_string())
+            ));
+        }
+    }
+
+    Some(context.snippet(span).to_owned())
```

Comment:
Could you make sure the values will continue to be passed through `wrap_str` as they are currently? If we drop that then we're potentially introducing some breaking formatting changes around some edge cases approaching width limits